server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -43,6 +48,7 @@ class Pipeline {
     private final List<SearchPhaseResultsProcessor> searchPhaseResultsProcessors;
     private final NamedWriteableRegistry namedWriteableRegistry;
     private final LongSupplier relativeTimeSupplier;
+    private int processorFailureCount = 0;
```

Comment:
We count failures in the metrics. See the `PipelineWithMetrics` class. 

All we need to do is invoke `.failed()` on the relevant metrics, as we do in the implementation of `onRequestProcessorFailed` and `onResponseProcessorFailed`.